### ISSUE TITLE
Is training support when using `device_map` and/or `low_cpu_mem_usage`

### ISSUE THREAD

**Issue by HanGuo97** [huggingface/accelerate#1805]
In the documentation page (such as [this](https://huggingface.co/docs/accelerate/usage_guides/big_modeling) and [that](https://huggingface.co/docs/transformers/main/main_classes/model#large-model-loading)), it is stated that `device_map` and `low_cpu_mem_usage` can only be used for inference, not training. The reason seems to be that these methods wrap around a couple `torch.no_grad` behind the scenes. Presumably this means gradients won't be available.

However, I noticed that, in practice, these methods still provide gradients as long as the model is fully on (maybe multiple) GPUs. Notably, the hooks have the attribute `no_grad` that was set to False by [default](https://github.com/huggingface/accelerate/blob/main/src/accelerate/hooks.py#L41) and never set to True [in practice](https://github.com/search?q=repo%3Ahuggingface%2Faccelerate%20%22.no_grad%20%3D%20True%22&type=code).

Am I correct in that as long as the models live fully on (one or many) GPUs, `device_map` and/or `low_cpu_mem_usage` in fact do support training (i.e., getting gradients)? (They might not be performant, but works.) Thanks in advance for the help!

A similar issue has been mentioned in the [forum](https://discuss.huggingface.co/t/using-device-map-auto-for-training/46321) but the answer does not seem particularly satisfying to me.

**Comment by sgugger** [comment]
Training is supported if your model is on multiple GPUs yes. It is only the offload to CPU/disk that ends up with training not supported.

**Comment by muellerzr** [comment]
I'll be writing some docs on this soon, great question @HanGuo97

**Comment by HanGuo97** [comment]
Got it -- thanks both for the clarification!
